### PR TITLE
Live activities push to start

### PR DESCRIPTION
**Pull request by jerielng**
# Pull Request/Issue Resolution

#### Description of Change:
Updated documentation to support the new Live Activities push-to-start features in iOS 17.2 (pending release)

#### Is this change associated with a Braze feature/product release?
- [X] Yes (**TBD**)
- [ ] No

---

<details>
<summary>✔️ Pull Request Checklist</summary>
<br>

- [ ] Check that you haven't removed any images (replacing an image with an updated one of the same name is fine), as this breaks the French site
- [ ] Check that all links work.
- [ ] Ensure you have completed [our Contributors License Agreement](https://www.braze.com/docs/cla/).
- [ ] Tag @josh-mccrowell-braze and @internetisaiah as a reviewer when your work is **done and ready to be reviewed for merge**. Are you an internal product manager? Reference the internal reviewing chart to tag the appropriate reviewer.
- [ ] If the documentation involves a 1) paid SKU, 2) a third party, 3) SMS, 4) AI, or 5) privacy, ensure that Maria Maldonado on the Legal team has signed off.
- [ ] Tag others as reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `assets` > `js` > `broken_redirect_list.js`

</details>

<details>
<summary>⭐ Helpful Wiki Shortcuts</summary>
<br>

- [Writing Style Guide](https://docs.google.com/document/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub)
- [Image Style Guide](https://docs.google.com/document/d/e/2PACX-1vRJSkwcjmjrTfLDagZccLpOMMyh5NN5SXRZSjz12cRAHbX4OrUmhvCmYpf_p5YB-9r4_jSOQLkicQIH/pub)
- [Styling Test Page](https://www.braze.com/docs/home/styling_test_page/)

</details>

<details>
<summary>❗ ATTN: For PR Reviewers</summary>
<br>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Read our [Previewing Documentation page](https://github.com/braze-inc/braze-docs/wiki/Previewing-and-Testing-Documentation) to see how to check the deployment.
  - [ ] Preview all changes in the linked Vercel environment by clicking the preview link in the vercel-bot comment in your PR.
</details>

<details>
<summary>❗ ATTN: Internal Reviewing Chart </summary>
<br>
<b>Work at Braze and not sure who to tag for review?</b> <br>Before tagging @josh-mccrowell-braze or @internetisaiah for a general review, reference the following chart to see if a specific product vertical/reviewer applies to your pull request.
<br><br>
<table>
<tr>
    <td><b>Reviewer</b></td>
    <td><b>Product Vertical</b></td>
  </tr>
  <tr>
    <td>@josh-mccrowell-braze</td>
    <td>Monolith Deployments<br>Quality Infrastructure<br>Platform Infrastructure<br>Datalake<br>SDKs<br>Currents</td>
  </tr>
  <tr>
    <td>@internetisaiah</td>
    <td>Developer Guide<br>Technology partners<br>Channels</td>
  </tr>
  <tr>
    <td>@bre-fitzgerald</td>
    <td>Intelligence<br>In-App Messages<br>Channels<br>Frontend Infrastructure & Experience (FIX)</td>
  </tr>
  <tr>
    <td>@lydia-xie</td>
    <td>Ingestion<br>Core Objects<br>Core Messaging<br>Messaging Experience<br>Message Components<br>Email (Composition and Infrastructure) (tag @rachel-feinberg for Liquid use cases)</td>
  </tr>
  <tr>
    <td>@rachel-feinberg</td>
    <td>Customer Lifecycle, Identity and Permissions<br>SMS<br>User Targeting<br>Reporting</td>
  </tr>
</table>
</details>

### CODE REVIEW

**Comment by jerielng** [comment]
Sorry @josh-mccrowell-braze, @internetisaiah - still a WIP, I mainly created this PR to get the deployment link to preview it, but let me know if there's a more streamlined process I can do without prematurely opening a PR!

**Comment by josh-mccrowell-braze** [comment]
@jerielng No, no this is good! I just tagged Isaiah because I know he's also working on this ticket and wanted him to be aware. It was just an FYI. We're standing by until you're done.

**Comment by internetisaiah** [comment]
@jerielng , nope this is perfect! I think Josh just added it to me so you know who to tag when the time comes 👍🏽  As you were 🖖🏽 (there's no regular salute emoji??)

File: _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md [hunk 1]
```diff
@@ -35,11 +37,15 @@ To manage the lifecycle of a Live Activity, follow these four steps.
 
 1. [Create the Live Activity.](#developing) Develop the Live Activity UI using WidgetKit and SwiftUI. Initialize a Live Activity object with the relevant data models for your static and dynamic states.<br><br>
 
-2. [Register the Live Activity](#registering) Register a Live Activity with the Braze SDK using the [`launchActivity`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/liveactivities-swift.class/launchactivity(pushtokentag:activity:fileid:line:)) method with the Live Activity object and unique activity tag.<br><br>
+2. Start the Live Activity.<br><br>
+   a. [via push-to-start](#starting) Register a Live Activity type with the Braze SDK using the [`registerPushToStart`](<https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/liveactivities-swift.class/launchactivity(pushtokentag:activity:fileid:line:)>) method and remotely start an activity of that type using our [`/messages/live_activity/start` endpoint]({{site.baseurl}}/api/endpoints/messaging/live_activity/start).<br><br>
```
Comment (jerielng) [inline _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md:41]: - [ ] Update this link after releasing the updated docs:
```suggestion
   a. [via push-to-start](#starting) Register a Live Activity type with the Braze SDK using the [`registerPushToStart`](<https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/liveactivities-swift.class/update-method>) method and remotely start an activity of that type using our [`/messages/live_activity/start` endpoint]({{site.baseurl}}/api/endpoints/messaging/live_activity/start).<br><br>
```

File: _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md [hunk 2]
```diff
@@ -78,9 +84,81 @@ struct BrazeActivityAttributes: ActivityAttributes {
 }
 ```
 
-## Step 2: Registering a Live Activity {#registering}
+## Step 2a: Registering and Starting a Live Activity remotely {#starting}
+
+{% alert note %}
+Live Activity push-to-start features require iOS 17.2+ and version 8.Y.Z of the Braze Swift SDK.
```
Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md:90]: todo: update after version is finalized

File: _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md [hunk 3]
```diff
@@ -22,11 +22,13 @@ As a developer, you can use Braze to manage your Live Activity lifecycles, make
 {% sdk_min_versions swift:5.11.0 %}
 
 Additional prerequisites include:
-* Live Activities are only available for iPhones on iOS 16.1 and later. To use this feature, ensure that your project is targeting this iOS version.
-* The `Push Notification` entitlement must be added under **Signing & Capabilities** in your Xcode project.
+
+- Live Activities are only available for iPhones and iPads on iOS 16.1 and later. To use this feature, ensure that your project is targeting this iOS version.
+- The `Push Notification` entitlement must be added under **Signing & Capabilities** in your Xcode project.
+- Remotely starting a Live Activity via push notifications was introduced in version 8.Y.Z of the Braze SDK and requires iOS 17.2 and later.
```
Comment (jerielng) [inline _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md:28]: - [ ] Update version
```suggestion
- Remotely starting a Live Activity via push notifications was introduced in version 8.1.0 of the Braze SDK and requires iOS 17.2 and later.
```

File: _docs/_developer_guide/platform_integration_guides/swift/live_activities/faq.md [hunk 4]
```diff
@@ -60,6 +60,10 @@ The API keys you use need to be given the correct permissions to access the diff
 
 The `messages/live_activity/update` endpoint has a separate rate limit from any other Braze endpoint. By default, the rate limit for the `messages/live_activity/update` endpoint is 250,000 requests per hour per workspace. See the [rate limit article][5] for more information.
 
+### Why aren't my push-to-start tokens being generated?
+
+As of iOS 17.2+, there are certain limitations in Apple's `pushToStartToken` and `pushToStartTokenUpdates` APIs. Push-to-start tokens are only generated on the first app launch in `application(_:didFinishLaunchingWithOptions:)` after the first install. If this step needs to be repeated, tokens can only be generated again after rebooting and re-installing the app, or by manually creating a new instance of that Live Activity.
```
Comment (hokstuff) [inline _docs/_developer_guide/platform_integration_guides/swift/live_activities/faq.md:65]: Optional nit: If we want to be more cautious with wording, we can use:
```suggestion
As of iOS 17.2+, there are certain limitations in Apple's `pushToStartToken` and `pushToStartTokenUpdates` APIs. In practice, push-to-start tokens are only generated on the first app launch in `application(_:didFinishLaunchingWithOptions:)` after the first install. If this step needs to be repeated, tokens can only be generated again after rebooting and re-installing the app, or by manually creating a new instance of that Live Activity.
```

**Review by hokstuff — APPROVED** [review]
LGTM!

**Review by davidbielik — APPROVED** [review]

**Review by jacksonemiller — APPROVED** [review]
Is this formatting expected?

Screenshot 2024-03-12 at 4 55 49 PM

**Comment by jerielng** [comment]
> Is this formatting expected?
> 
> Screenshot 2024-03-12 at 4 55 49 PM

Looks a bit off, although that seems to be on our [published docs](https://www.braze.com/docs/api/endpoints/messaging/live_activity/update/#example-error-response) for the existing `/update` endpoint as well. @internetisaiah Do you know if this is expected, or is there a way to make it display better?

**Comment by internetisaiah** [comment]
> > Is this formatting expected?
> > Screenshot 2024-03-12 at 4 55 49 PM
> 
> Looks a bit off, although that seems to be on our [published docs](https://www.braze.com/docs/api/endpoints/messaging/live_activity/update/#example-error-response) for the existing `/update` endpoint as well. @internetisaiah Do you know if this is expected, or is there a way to make it display better?

@jerielng , follow up here:

After chatting with the site maintenance engineers, the issue seems to be that "newline characters are technically invalid json." That being said, given the context of this doc, it seems unlikely that we can just change this code block since it's meant to be a representation of what end-users will see returned when using this API endpoint.

If this is just the standard way _all_ our endpoints work, then I can put in a request (or try myself) to update our CSS handling rules for this kind of syntax highlighting.

What are your thoughts?

**Comment by davidbielik** [comment]
> > > Is this formatting expected?
> > > Screenshot 2024-03-12 at 4 55 49 PM
> > 
> > 
> > Looks a bit off, although that seems to be on our [published docs](https://www.braze.com/docs/api/endpoints/messaging/live_activity/update/#example-error-response) for the existing `/update` endpoint as well. @internetisaiah Do you know if this is expected, or is there a way to make it display better?
> 
> @jerielng , follow up here:
> 
> After chatting with the site maintenance engineers, the issue seems to be that "newline characters are technically invalid json." That being said, given the context of this doc, it seems unlikely that we can just change this code block since it's meant to be a representation of what end-users will see returned when using this API endpoint.
> 
> If this is just the standard way _all_ our endpoints work, then I can put in a request (or try myself) to update our CSS handling rules for this kind of syntax highlighting.
> 
> What are your thoughts?

yea this is expected and can be found in other parts of docs today.

**Comment by internetisaiah** [comment]
> @internetisaiah One other question, do you know who owns this [page](https://documenter.getpostman.com/view/4689407/SVYrsdsG?version=latest#2300226e-f26a-4154-9bcc-5883f1f294cd)? There's a link to it from the new `/start`[ endpoint docs](https://braze-docs-6mmr8g3p1-braze.vercel.app/docs/api/endpoints/messaging/live_activity/start), but it still points to our samples for `/update` because it doesn't exist yet.

We own it! And I can get it updated. Thanks for the heads up ✅

**Comment by davidbielik** [comment]
hey @internetisaiah any chance this can be published today?

**Comment by internetisaiah** [comment]
@davidbielik , ah yes! I can review and merge today! I was thinking this would be released with the other new-feature related docs next week for push to start. I should have clarified previously. Apologies!

**Comment by internetisaiah** [comment]
Hey @davidbielik , I still need to clean up the images and do a pass on the FAQ. In the meantime, can you or someone on the team checkout the changes I made in my latest commit?

I decided to take out the "overview" section towards the top of the **Implementing a Live Activity** section as it didn't feel that it added to the overall doc.

That being said, I may be overlooking an important use case so let me know if thats alright!

File: _docs/_api/endpoints/messaging/live_activity/update.md [hunk 5]
```diff
@@ -17,15 +17,16 @@ description: "This article outlines details about the Update Live Activity endpo
...
 
 {% apiref postman %}https://documenter.getpostman.com/view/4689407/SVYrsdsG?version=latest#2300226e-f26a-4154-9bcc-5883f1f294cd {% endapiref %}
 
 ## Prerequisites
 
-To use this endpoint, you'll need to generate an API key with the `messages.live_activity.update` permission.
+To use this endpoint, you'll need to complete the following:
+
+- Generate an API key with the `messages.live_activity.update` permission.
+- Register a Live Activity [remotely]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/live_activities/live_activities/?tab=remote#step-2-register-the-activity) or [locally]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/live_activities/live_activities/?tab=local#step-2-register-the-activity) using the Braze Swift SDK.
```
Comment (davidbielik) [inline _docs/_api/endpoints/messaging/live_activity/update.md:29]: is this the right link?

File: _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md [hunk 6]
```diff
@@ -78,11 +70,86 @@ struct BrazeActivityAttributes: ActivityAttributes {
 }
 ```
 
-## Step 2: Registering a Live Activity {#registering}
+### Step 2: Register the activity
```
Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md:73]:
```suggestion
### Step 2: Start the activity
```

@jerielng would it make sense to call this start, or is register the proper terminology here.

File: _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md [hunk 7]
```diff
@@ -78,11 +70,86 @@ struct BrazeActivityAttributes: ActivityAttributes {
...
 
-Next, you will use Braze methods to track and manage your Live Activities. 
+First, choose how you want to register your activity:
 
-Updates to a Live Activity can be sent using ActivityKit (Apple’s framework for managing a Live Activity) or remote push notifications. In this instance, you will use ActivityKit to get a push token, which the Braze SDK can manage for you. This allows you to update Live Activities through the Braze API, as Braze will send the push token to the Apple Push Notification service (APNs) on the backend.
+- **Remote:** Use the [`registerPushToStart`](http://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/liveactivities-swift.class/registerpushtostart(fortype:name:)) method, then start an activity using the [`/messages/live_activity/start`]({{site.baseurl}}/api/endpoints/messaging/live_activity/start) endpoint.
```
Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md:77]:
```suggestion
- **Remotely:** Use the [`registerPushToStart`](http://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/liveactivities-swift.class/registerpushtostart(fortype:name:)) method, then start an activity using the [`/messages/live_activity/start`]({{site.baseurl}}/api/endpoints/messaging/live_activity/start) endpoint.
```

File: _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md [hunk 8]
```diff
@@ -78,11 +70,86 @@ struct BrazeActivityAttributes: ActivityAttributes {
...
+- **Local:** Use the [`launchActivity`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/liveactivities-swift.class/launchactivity(pushtokentag:activity:fileid:line:)) method to create push tokens for the Braze SDK to manage.
```
Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md:78]:
```suggestion
- **Locally:** Use the [`launchActivity`](https://braze-inc.github.io/braze-swift-sdk/documentation/brazekit/braze/liveactivities-swift.class/launchactivity(pushtokentag:activity:fileid:line:)) method to create push tokens for the Braze SDK to manage.
```

just a suggestion (here and tabs below)

File: _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md [hunk 9]
```diff
@@ -91,9 +158,9 @@ Updates to a Live Activity can be sent using ActivityKit (Apple’s framework fo
 
 Once you have registered the Live Activity, the Braze SDK will extract and observe changes in the push tokens.
 
-### Example
+#### Example
 
-For our example, we’ll create class called LiveActivityManager as an interface for our Live Activity objects. Then, we'll set the `pushTokenTag` to `"live-activity-1"`.
+For our example, we’ll create class called `LiveActivityManager` as an interface for our Live Activity objects. Then, we'll set the `pushTokenTag` to `"live-activity-1"`.
```
Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md:163]:
```suggestion
For our example, we’ll create class called `LiveActivityManager` as an interface for our Live Activity objects. Then, we'll set the `pushTokenTag` to `"sports-game-2024-03-15"`.
```

would be good to have a clear example activity ID like this

**Review by davidbielik — APPROVED** [review]
few suggestions, but LGTM!

File: _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md (hunk 6 above)
Comment (jerielng) [reply _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md:73]: Yep I feel like `Start` would be better, although we'd need to make a couple of changes to the content. I'll take a dive into that.

File: _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md [hunk 10]
```diff
@@ -106,16 +173,18 @@ class LiveActivityManager {
...
                                             content: activityContent,
       // Setting your pushType as .token allows the Activity to generate push tokens for the server to watch.
                                             pushType: .token) {
-      // Register your Live Activity with Braze using the pushTokenTag
-      AppDelegate.braze?.liveActivities.launchActivity(pushTokenTag: "live-activity-1",
-                                                       activity: activity)
+      // Register your Live Activity with Braze using the pushTokenTag.
+      // This method returns a Swift background task.
+      // You may keep a reference to this task if you need to cancel it wherever appropriate, or ignore the return value if you wish.
+      let liveActivityObserver: Task = AppDelegate.braze?.liveActivities.launchActivity(pushTokenTag: "live-activity-1",
```
Comment (jerielng) [inline _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md:186]:
```suggestion
      let liveActivityObserver: Task = AppDelegate.braze?.liveActivities.launchActivity(pushTokenTag: "sports-game-2024-03-15",
```

File: _docs/_api/endpoints/messaging/live_activity/update.md (hunk 5 above)
Comment (jerielng) [reply _docs/_api/endpoints/messaging/live_activity/update.md:29]: It's correct - looks like they just link to the same section, which has two paths

**Comment by jerielng** [comment]
@internetisaiah I made a few minor adjustments based on some of the feedback from @davidbielik if you could take a second look to see if anything I changed could be phrased better, thanks!

File: _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md [hunk 11]
```diff
@@ -78,11 +70,86 @@ struct BrazeActivityAttributes: ActivityAttributes {
...
+
+}
+```
+{% endtab %}
+
+{% tab local %}
+You can use [Apple's ActivityKit framework](https://developer.apple.com/documentation/activitykit) to get a push token, which the Braze SDK can manage for you. This allows you to update Live Activities through the Braze API, as Braze will send the push token to the Apple Push Notification service (APNs) on the backend.
 
 1. Create an instance of your Live Activity implementation using Apple’s ActivityKit APIs.
 2. Set the `pushType` parameter as `.token`.
```
[the remark this answers is not in the document]
Comment (internetisaiah) [reply _docs/_developer_guide/platform_integration_guides/swift/live_activities/live_activities.md:160]: Good catch! Yes, for some reason there can be issues when trying to applying markdown styling like `**` inside a link + using reference-styling linking. So changing it to this instead, fixed the issue: `[text](url)`.

**Review by internetisaiah — APPROVED** [review]
LGTM

**Pull request merged**